ext/soap/php_http.c

```diff
@@ -639,11 +639,15 @@ int make_http_soap_request(zval        *this_ptr,
 			if ((digest = zend_hash_str_find(Z_OBJPROP_P(this_ptr), "_digest", sizeof("_digest")-1)) != NULL) {
 				if (Z_TYPE_P(digest) == IS_ARRAY) {
 					char          HA1[33], HA2[33], response[33], cnonce[33], nc[9];
+					zend_long     nonce;
 					PHP_MD5_CTX   md5ctx;
 					unsigned char hash[16];
 
+					php_random_bytes_throw(&nonce, sizeof(nonce));
+					nonce &= 0x7fffffff;
```

Comment:
this suggests that `sizeof(nonce)` is guaranteed to be 4. the previous suggests that it is not